### PR TITLE
Move the repo restore to static graph

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes:https://github.com/NuGet/Client.Engineering/issues/790

Regression? Last working version:

## Description

It's a good idea to dog food our own features :)
This saves us 3-5s depending on the machine.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Comment by erdembayar** [comment]
@nkolev92 
Can you trigger the tests again? Before I approve this I want to see if it passes all the tests.

**Comment by nkolev92** [comment]
I think you might be onto something, this change affects some of tests that run `.test` folder. 


I'll wait for the results, but I'm pretty sure it'll need some extra tweaking (probably empty `Directory.Build.Props`/`Directory.Build.Targets`)

File: Directory.Solution.Props [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<Project>
+    <Import Project="Directory.Build.Props" />
```
Comment (Nirmal4G) [inline Directory.Solution.Props:2]: Prefer lower-case file extension instead of Title-case. e.g.: `Directory.Build.props`

Comment (nkolev92) [reply Directory.Solution.Props:2]: The casing matters cross platform.
I incorrectly remembered what the xplat behavior is.
`props` and `targets` are what MSBuild looks for. 

Nice catch.

**Comment by nkolev92** [comment]
This needs some extra work. 
I'll pick it up later, closing for now.

**Pull request closed without merging**